Automattic/vip-design-system
# Add new Toolbar component

## Description

This component is the base for the Responsiveness Project Header Part. I will create at least two more components:

- Toolbar.Toggler (to create a hamburger menu and open the Navigation sidebar)
- Navigation sidebar (Migration from the VIP Dashboard to a generic component)

### What's included:

- Generation of the correct Valet Theme Breakpoints built-in with theme-ui;
- The new Toolbar (former Header) component and its sub-parts;
- A new variant for the `Nav` component, allowing us to re-use code 🥳 . The Toolbar can now use the `Nav` component
- Storybook examples;
- Unit tests

<img width="1026" alt="image" src="https://github.com/Automattic/vip-design-system/assets/3402/2e66a0c9-768d-4031-8f57-41a10f0dacb8">

<img width="1034" alt="image" src="https://github.com/Automattic/vip-design-system/assets/3402/450355b5-e6e8-4b81-a442-f812af170892">


### The Plan

- After merging this component, I will put it in the VIP Dashboard. Once the other parts are done, we will move the other parts to the VIP Dashboard, including the Responsiveness behavior

## Checklist


## Steps to Test

1. Pull down PR. OR go to https://deploy-preview-298--vip-design-system-components.netlify.app/ and jump to item 3
2. `npm run dev`.
3. Open Storybook in Navigation / Toolbar
4. Check all components of Stories
5. Navigation works with the keyboard
6. The component is similar to the VIP Dashboard header component
